vcs/repository.go

```diff
@@ -140,10 +141,28 @@ type Diff struct {
 	Raw string // the raw diff output
 }
 
+// BranchesOptions specifies options for the list of branches returned by
+// (Repository).Branches.
+type BranchesOptions struct {
+	// BehindAheadBranch specifies a branch name. If set to something other than blank string,
+	// then each returned branch will include a behind/ahead commit counts information
+	// against the specified base branch. If left blank, then branches
+	// will not include that information and their Counts will be nil.
+	BehindAheadBranch string `json:",omitempty" url:",omitempty"`
+}
+
 // A Branch is a VCS branch.
 type Branch struct {
 	Name string
 	Head CommitID
+
+	Counts *BehindAhead `json:",omitempty"` // Counts optionally contains the commit counts relative to specified branch.
+}
+
+// BehindAhead is a set of behind/ahead counts.
+type BehindAhead struct {
+	Behind int
+	Ahead  int
```

Comment:
Yeah, that would make it clearer actually. I notice that the Go stdlib uses signed ints in many places where the values are constrained to be nonnegative--perhaps to avoid type conversions. But I am all for additional clarity.